core/src/main/java/com/valtech/aem/saas/core/tracking/SearchResultItemTrackingServlet.java

```diff
@@ -0,0 +1,82 @@
+package com.valtech.aem.saas.core.tracking;
+
+import com.valtech.aem.saas.api.fulltextsearch.SearchModel;
+import com.valtech.aem.saas.api.tracking.TrackingService;
+import com.valtech.aem.saas.api.tracking.dto.SearchResultItemTrackingDTO;
+import com.valtech.aem.saas.core.common.request.RequestWrapper;
+import com.valtech.aem.saas.core.fulltextsearch.SearchModelImpl;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+import org.apache.sling.servlets.annotations.SlingServletResourceTypes;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Optional;
+
+import static com.valtech.aem.saas.core.tracking.SearchResultItemTrackingServlet.SEARCH_RESULT_ITEM_TRACKING_EXTENSION;
+
+@Slf4j
+@Component(service = Servlet.class)
+@SlingServletResourceTypes(resourceTypes = SearchModelImpl.RESOURCE_TYPE,
+                           methods = HttpPost.METHOD_NAME,
+                           selectors = SearchResultItemTrackingServlet.SEARCH_RESULT_ITEM_TRACKING_SELECTOR,
+                           extensions = SEARCH_RESULT_ITEM_TRACKING_EXTENSION)
+public class SearchResultItemTrackingServlet extends SlingAllMethodsServlet {
+
+    public static final String SEARCH_RESULT_ITEM_TRACKING_SELECTOR = "tracking";
+    public static final String SEARCH_RESULT_ITEM_TRACKING_EXTENSION = "html";
```

Comment:
- [ ] as we return a json I think we should change to json extension. What do you think?

Reply:
aha, we could also just return the response from saas as a json, no? gives some more context in case of an issue